lib/Sema/CSSolver.cpp

```diff
@@ -991,8 +991,13 @@ void ConstraintSystem::shrink(Expr *expr) {
           auto *const typeRepr = coerceExpr->getCastTypeRepr();
 
           if (typeRepr && isSuitableCollection(typeRepr)) {
-            auto resolution = TypeResolution::forContextual(CS.DC, None);
-            auto coercionType = resolution.resolveType(typeRepr);
+            const auto coercionType =
+                TypeResolution::forContextual(
+                    CS.DC, None,
+                    // FIXME: Should we really unconditionally complain
+                    // about unbound generics here?
```

Comment:
if they're truly invalid, then instead of complaining, we should be ensuring that the TypeResolution's context kind and so on doesn't allow them in the first place.

Reply:
Perhaps the sentence is flawed. What I meant to convey here is that we may want to someday reconsider complaining about unbound generics here (we *do* complain today). For example:
```swift 
let foo: [Array<Float>] = [[3], [4], [5]] as [Array] // error
```